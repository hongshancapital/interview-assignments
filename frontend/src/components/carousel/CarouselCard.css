.carousel-card {
  width: 100%;
  height: 100%;
  box-sizing: border-box;
  background-repeat: no-repeat;
  background-size: auto 50%;
  background-position: center 35vh;
  overflow: hidden;
  white-space: nowrap;
}

.carousel-card__title {
  margin: 25vh 0 10px;
  font-size: 56px;
  font-weight: 600;
  white-space: pre-wrap;
}

.carousel-card__desc {
  margin: 0;
  font-size: 28px;
  font-weight: 400;
  white-space: pre-wrap;
}
