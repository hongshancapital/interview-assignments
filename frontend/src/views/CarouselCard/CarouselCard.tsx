import { PropsWithChildren } from 'react';
import styles from './styles/CarouselCard.module.scss';

const CarouselCard: React.FC<
  PropsWithChildren<{ backgroundColor: string; backgroundImage: string }>
> = ({ backgroundColor, backgroundImage, children }) => (
  <div
    className={styles.container}
    style={{
      backgroundColor,
      backgroundImage: `url(${backgroundImage})`
    }}
  >
    {children}
  </div>
);

export default CarouselCard;
