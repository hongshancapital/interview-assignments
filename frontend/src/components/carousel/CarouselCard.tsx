import './CarouselCard.css';

export interface CarouselCardProps {
  className?: string;
  title: string;
  desc?: string;
  bgImage: string;
  color: string;
  bgColor: string;
}

const CarouselCard = ({ className, title, desc, bgImage, color, bgColor }: CarouselCardProps) => {
  return (
    <div
      className={['carousel-card', className].filter(Boolean).join(' ')}
      style={{
        color,
        backgroundColor: bgColor,
        backgroundImage: `url(${bgImage})`,
      }}
    >
      <h2 className={'carousel-card__title'}>{title}</h2>
      {desc && <p className={'carousel-card__desc'}>{desc}</p>}
    </div>
  );
};
CarouselCard.displayName = 'CarouselCard';

export default CarouselCard;
