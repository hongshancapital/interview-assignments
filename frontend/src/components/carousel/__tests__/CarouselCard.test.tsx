import { render } from '@testing-library/react';
import CarouselCard from '../CarouselCard';

describe('CarouselCard', () => {
  it('should render CarouselCard component', () => {
    const { asFragment } = render(
      <CarouselCard
        title="title"
        desc="desc"
        color="#fff"
        bgColor="#fff"
        bgImage="image url"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
