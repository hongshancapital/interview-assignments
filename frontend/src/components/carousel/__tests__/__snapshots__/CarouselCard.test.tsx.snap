// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarouselCard should render CarouselCard component 1`] = `
<DocumentFragment>
  <div
    class="carousel-card"
    style="color: rgb(255, 255, 255); background-color: rgb(255, 255, 255);"
  >
    <h2
      class="carousel-card__title"
    >
      title
    </h2>
    <p
      class="carousel-card__desc"
    >
      desc
    </p>
  </div>
</DocumentFragment>
`;
